tests/accumulator/test_use_cases.py

```diff
@@ -0,0 +1,212 @@
+
+
+class StreamSorterAccumulator(Accumulator):
+    """
+    Accumulator that sorts events by event_time and watermark.
+    This demonstrates custom sorting use case.
+    """
+
+    def __init__(self):
+        self.buffer: list[Datum] = []
+
+    async def handler(self, datums: AsyncIterable[Datum], output: NonBlockingIterator):
+        # Collect all datums
+        datum_count = 0
+        async for datum in datums:
+            datum_count += 1
+            self.buffer.append(datum)
+
+        # Sort by event_time
+        self.buffer.sort(key=lambda d: d.event_time)
+
+        # Emit sorted datums
+        message_count = 0
+        for datum in self.buffer:
+            message_count += 1
+            await output.put(
+                Message(
+                    value=datum.value,
+                    keys=datum.keys(),
+                    tags=[],
+                )
+            )
+
+        # Clear buffer for next window
+        self.buffer.clear()
+
+
+class TestAccumulatorUseCases(unittest.TestCase):
+    """Test practical accumulator use cases that developers can reference."""
+
+    def test_stream_sorter_end_to_end_example(self):
+        """Complete end-to-end example showing how to build a stream sorting accumulator.
+
+        This test demonstrates:
+        1. How to implement the Accumulator abstract class
+        2. How to process async streams of Datum objects
+        3. How to emit results as Message objects
+        4. Proper resource management (buffer clearing)
+        """
+
+        async def _test_end_to_end():
+            # Create accumulator instance
+            sorter = StreamSorterAccumulator()
+            output = NonBlockingIterator()
+
+            # Create test data - intentionally out of chronological order
+            test_data = [
+                create_test_datum(
+                    keys=["sensor_data"],
+                    value=b"temperature:25.5",
+                    event_time=datetime.fromtimestamp(1662998460, timezone.utc),  # Latest
+                    id_="temp_3",
+                ),
+                create_test_datum(
+                    keys=["sensor_data"],
+                    value=b"temperature:22.1",
+                    event_time=datetime.fromtimestamp(1662998400, timezone.utc),  # Earliest
+                    id_="temp_1",
+                ),
+                create_test_datum(
+                    keys=["sensor_data"],
+                    value=b"temperature:23.8",
+                    event_time=datetime.fromtimestamp(1662998430, timezone.utc),  # Middle
+                    id_="temp_2",
+                ),
+            ]
+
+            # Create async generator (simulates streaming data)
+            async def data_stream():
+                for datum in test_data:
+                    yield datum
+
+            # Process the stream through our accumulator
+            await sorter.handler(data_stream(), output)
```

Comment:
This function is draining the iterator and sending a sorted order. Not sure if this covers accumulator functionality here much as we are calling the function directly without any RPC interations. 

I guess this file can be removed in favor of actual end to end tests if that is the intention here